sympy/physics/mechanics/tests/test_eombase.py

```diff
@@ -0,0 +1,30 @@
+import sympy.physics.mechanics.eombase as eombase
+
+# Determine what inputs should be used
+EOM = eombase.EOMBase(frame=None)
```

Comment:
We currently have `LagrangesMethod` and `KanesMethod` classes. These classes take descriptions of multibody dynamic systems in the form of generalized coordinates/speeds, kinematically defined bodies, and constraints and produce symbolic equations of motion, i.e. it automates the EoM formation method.

We'd like to support future methods, e.g. `HamiltonMethod`, `NewtonEulerMethod`, `FeatherstoneMethod`, `DenavitHartenbergMethod` etc. We'd also like to support any general "manual" method. This could mean that the user finds the EoMs written already for them, or that they write them out by hand and then type them in SymPy code, or that they use SymPy to do symbolic computations for their custom method. The results of any of these should be everything that fully describes the symbolic form of the equations of motion.

One thought: Should we have a class that describes "equations of motion", and if so should the `*Method` classes produce an instance of this?

One thing that is nice about the current `*Method` classes is that they contain references to the bodies and loads. Should we care about that?

Reply:
Hello,
I have gone through the gsoc 2020 ideas list and was interested in creating a new class for NewtonEuler's method and Hamilton's method. I am new to Sympy's codebase. I have already gone through Sympy's tutorial and the Introduction to contributing page. Could you suggest me on how could I start to contribute.
Thank you.

Reply:
Start here: https://github.com/sympy/sympy/wiki/introduction-to-contributing

Reply:
Thank you very much.